src/api/v4/layer/layer.js

```diff
@@ -75,8 +77,9 @@ function Layer (source, style, options) {
   this._visible = true;
   this._featureClickColumns = options.featureClickColumns || [];
   this._featureOverColumns = options.featureOverColumns || [];
-
-  Base.apply(this, arguments);
+  if (options.id) {
+    this._id = options.id;
+  }
```

Comment:
With this approach, we generate an `id` always and then, in case the user provides an ID, it gets overridden.

I think it'd better to pass the `id` to `Base` as an option and, if present, assign it and avoiding generating one. `Base` exists only to manage the `id` and the `internalModel` so it seems a good place to contain all logic related to `ids`.

Reply:
good catch